### PR TITLE
Tinygorush

### PR DESCRIPTION
**Pull request by rminnich**

### CODE REVIEW

**Comment by rminnich** [comment]
so here is the POC

this runs the commands without using fork. That means it can work on bare metal environments like tinygo and tamago

It has a very simple problem:
/tmp/tiny3283520552/builder407189277/src/bb.u-root.com/bb/bb rush
% date
2023/02/26 00:46:00 run /usr/bin/date
Sun Feb 26 00:46:00 UTC 2023
% rush
2023/02/26 00:46:03 run rush
% date
2023/02/26 00:46:05 run /usr/bin/date
/tmp/tiny3283520552/builder407189277/src/bb.u-root.com/bb/bb flag redefined: u
panic: /tmp/tiny3283520552/builder407189277/src/bb.u-root.com/bb/bb flag redefined: u

because of globals :-)

**Comment by rminnich** [comment]
ok, with the latest
% date
2023/02/26 01:03:15 run /usr/bin/date
Sun Feb 26 01:03:15 UTC 2023
% date
2023/02/26 01:03:16 run /usr/bin/date
Sun Feb 26 01:03:16 UTC 2023
% date
2023/02/26 01:03:17 run /usr/bin/date
Sun Feb 26 01:03:17 UTC 2023
% date
2023/02/26 01:03:17 run /usr/bin/date
Sun Feb 26 01:03:17 UTC 2023
% 

no more problems with flag.

File: tools/tinygobb/main.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	// the busybox. I was in a hurry.
+	err = filepath.WalkDir(build[0], func(path string, d fs.DirEntry, err error) error {
+		if d.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(path, ".go") {
+			return nil
+		}
+
+		b, err := os.ReadFile(path)
```
Comment (binjip978) [inline tools/tinygobb/main.go:102]: staticcheck complains here, you override err on line 94 here before it used, I would rename it to something like errRead

**Comment by binjip978** [comment]
Otherwise it works for me:
```
root@epiphyte:/tmp/tiny588138646/builder4051922320/src/bb.u-root.com/bb# ./bb rush
% date
2023/02/26 09:51:22 run /usr/bin/date
Sun Feb 26 09:51:22 UTC 2023
```

**Review by binjip978 — APPROVED** [review]

**Pull request merged**